Modules/vc-module-orders/VirtoCommerce.OrdersModule.Data/Services/DefaultCustomerOrderTotalsCalculator.cs

```diff
@@ -48,33 +48,42 @@ public virtual void CalculateTotals(CustomerOrder order)
                 }
             }
 
-            order.DiscountTotal = 0m;
-            order.DiscountTotalWithTax = 0m;
             order.FeeTotal = order.Fee;
-            order.TaxTotal = 0m;
 
             if (!order.Items.IsNullOrEmpty())
             {
-                order.SubTotal = order.Items.Sum(x => x.Price * x.Quantity);
-                order.SubTotalWithTax = order.Items.Sum(x => x.PriceWithTax * x.Quantity);
-                order.SubTotalTaxTotal += order.Items.Sum(x => x.TaxTotal);
-                order.SubTotalDiscount = order.Items.Sum(x => x.DiscountTotal);
-                order.SubTotalDiscountWithTax = order.Items.Sum(x => x.DiscountTotalWithTax);
-                order.DiscountTotal += order.Items.Sum(x => x.DiscountTotal);
-                order.DiscountTotalWithTax += order.Items.Sum(x => x.DiscountTotalWithTax);
-                order.FeeTotal += order.Items.Sum(x => x.Fee);
-                order.FeeTotalWithTax += order.Items.Sum(x => x.FeeWithTax);
-                order.TaxTotal += order.Items.Sum(x => x.TaxTotal);
+                var price = order.Items.Any(x => x.Price.HasValue) ? order.Items.Sum(x => x.Price * x.Quantity) : null;
```

Comment:
Isn't it the same as 
```
order.SubTotal = order.Items.Any(x => x.Price.HasValue) ? order.Items.Sum(x => x.Price * x.Quantity) : order.SubTotal;
```
Moreover - do we need to make `Convert.ToDecimal(order.SubTotal) + price` here, as there is just an assignment, not +=?

For SubTotalTaxTotal (+= case) we could use
```
order.SubTotalTaxTotal = order.Items.Any(x => x.SubTotalTaxTotal.HasValue) ? order.Items.Sum(x => x.TaxTotal) + Convert.ToDecimal(order.SubTotalTaxTotal) : order.SubTotalTaxTotal;
```
Need to revise all these cases.